diff -uNr qemu-6.1.0/block.c qemu-6.1.0.mod/block.c
--- qemu-6.1.0/block.c	2021-08-24 20:35:40.000000000 +0300
+++ qemu-6.1.0.mod/block.c	2022-02-02 14:14:47.621030257 +0200
@@ -830,14 +830,16 @@
             ? 0 : -GetLastError());
 #else
     int fd;
-    const char *tmpdir;
-    tmpdir = getenv("TMPDIR");
+    char *tmpdir = getenv("TMPDIR");
+    char *prefixed_tmpdir = android_dynamic_path_prefix("/tmp");
     if (!tmpdir) {
-        tmpdir = "/var/tmp";
+        tmpdir = prefixed_tmpdir;
     }
     if (snprintf(filename, size, "%s/vl.XXXXXX", tmpdir) >= size) {
+        free(prefixed_tmpdir);
         return -EOVERFLOW;
     }
+    free(prefixed_tmpdir);
     fd = mkstemp(filename);
     if (fd < 0) {
         return -errno;
diff -uNr qemu-6.1.0/hw/hppa/machine.c qemu-6.1.0.mod/hw/hppa/machine.c
--- qemu-6.1.0/hw/hppa/machine.c	2021-08-24 20:35:40.000000000 +0300
+++ qemu-6.1.0.mod/hw/hppa/machine.c	2022-02-02 14:14:34.168946230 +0200
@@ -100,19 +100,19 @@
     fw_cfg_add_i64(fw_cfg, FW_CFG_RAM_SIZE, ms->ram_size);
 
     val = cpu_to_le64(MIN_SEABIOS_HPPA_VERSION);
-    fw_cfg_add_file(fw_cfg, "/etc/firmware-min-version",
+    fw_cfg_add_file(fw_cfg, APPLICATION_RUNTIME_PREFIX "/etc/firmware-min-version",
                     g_memdup(&val, sizeof(val)), sizeof(val));
 
     val = cpu_to_le64(HPPA_TLB_ENTRIES);
-    fw_cfg_add_file(fw_cfg, "/etc/cpu/tlb_entries",
+    fw_cfg_add_file(fw_cfg, APPLICATION_RUNTIME_PREFIX "/etc/cpu/tlb_entries",
                     g_memdup(&val, sizeof(val)), sizeof(val));
 
     val = cpu_to_le64(HPPA_BTLB_ENTRIES);
-    fw_cfg_add_file(fw_cfg, "/etc/cpu/btlb_entries",
+    fw_cfg_add_file(fw_cfg, APPLICATION_RUNTIME_PREFIX "/etc/cpu/btlb_entries",
                     g_memdup(&val, sizeof(val)), sizeof(val));
 
     val = cpu_to_le64(HPA_POWER_BUTTON);
-    fw_cfg_add_file(fw_cfg, "/etc/power-button-addr",
+    fw_cfg_add_file(fw_cfg, APPLICATION_RUNTIME_PREFIX "/etc/power-button-addr",
                     g_memdup(&val, sizeof(val)), sizeof(val));
 
     fw_cfg_add_i16(fw_cfg, FW_CFG_BOOT_DEVICE, ms->boot_order[0]);
diff -uNr qemu-6.1.0/hw/usb/ccid-card-emulated.c qemu-6.1.0.mod/hw/usb/ccid-card-emulated.c
--- qemu-6.1.0/hw/usb/ccid-card-emulated.c	2021-08-24 20:35:40.000000000 +0300
+++ qemu-6.1.0.mod/hw/usb/ccid-card-emulated.c	2022-02-02 14:14:47.621030257 +0200
@@ -417,7 +417,7 @@
     event_notifier_cleanup(&card->notifier);
 }
 
-#define CERTIFICATES_DEFAULT_DB "/etc/pki/nssdb"
+#define CERTIFICATES_DEFAULT_DB "./pki/nssdb"
 #define CERTIFICATES_ARGS_TEMPLATE\
     "db=\"%s\" use_hw=no soft=(,Virtual Reader,CAC,,%s,%s,%s)"
 
diff -uNr qemu-6.1.0/linux-user/syscall.c qemu-6.1.0.mod/linux-user/syscall.c
--- qemu-6.1.0/linux-user/syscall.c	2021-08-24 20:35:41.000000000 +0300
+++ qemu-6.1.0.mod/linux-user/syscall.c	2022-02-02 14:14:47.625030282 +0200
@@ -8017,9 +8017,11 @@
 
         /* create temporary file to map stat to */
         tmpdir = getenv("TMPDIR");
+        char *prefixed_tmpdir = android_dynamic_path_prefix("/tmp");
         if (!tmpdir)
-            tmpdir = "/tmp";
+            tmpdir = prefixed_tmpdir;
         snprintf(filename, sizeof(filename), "%s/qemu-open.XXXXXX", tmpdir);
+        free(prefixed_tmpdir);
         fd = mkstemp(filename);
         if (fd < 0) {
             return fd;
diff -uNr qemu-6.1.0/migration/exec.c qemu-6.1.0.mod/migration/exec.c
--- qemu-6.1.0/migration/exec.c	2021-08-24 20:35:41.000000000 +0300
+++ qemu-6.1.0.mod/migration/exec.c	2022-02-02 14:14:47.625030282 +0200
@@ -28,7 +28,7 @@
 void exec_start_outgoing_migration(MigrationState *s, const char *command, Error **errp)
 {
     QIOChannel *ioc;
-    const char *argv[] = { "/bin/sh", "-c", command, NULL };
+    const char *argv[] = { "/system/bin/sh", "-c", command, NULL };
 
     trace_migration_exec_outgoing(command);
     ioc = QIO_CHANNEL(qio_channel_command_new_spawn(argv,
@@ -55,7 +55,7 @@
 void exec_start_incoming_migration(const char *command, Error **errp)
 {
     QIOChannel *ioc;
-    const char *argv[] = { "/bin/sh", "-c", command, NULL };
+    const char *argv[] = { "/system/bin/sh", "-c", command, NULL };
 
     trace_migration_exec_incoming(command);
     ioc = QIO_CHANNEL(qio_channel_command_new_spawn(argv,
diff -uNr qemu-6.1.0/net/tap.c qemu-6.1.0.mod/net/tap.c
--- qemu-6.1.0/net/tap.c	2021-08-24 20:35:41.000000000 +0300
+++ qemu-6.1.0.mod/net/tap.c	2022-02-02 14:14:47.629030307 +0200
@@ -552,7 +552,7 @@
             *parg++ = helper_cmd;
             *parg++ = NULL;
 
-            execv("/bin/sh", args);
+            execv("/system/bin/sh", args);
             g_free(helper_cmd);
         } else {
             /* assume helper is just the executable path name */
diff -uNr qemu-6.1.0/qemu-nbd.c qemu-6.1.0.mod/qemu-nbd.c
--- qemu-6.1.0/qemu-nbd.c	2021-08-24 20:35:41.000000000 +0300
+++ qemu-6.1.0.mod/qemu-nbd.c	2022-02-02 14:14:34.176946280 +0200
@@ -53,7 +53,6 @@
 #define HAVE_NBD_DEVICE 0
 #endif
 
-#define SOCKET_PATH                "/var/lock/qemu-nbd-%s"
 #define QEMU_NBD_OPT_CACHE         256
 #define QEMU_NBD_OPT_AIO           257
 #define QEMU_NBD_OPT_DISCARD       258
@@ -92,7 +91,7 @@
 "  -p, --port=PORT           port to listen on (default `%d')\n"
 "  -b, --bind=IFACE          interface to bind to (default `0.0.0.0')\n"
 "  -k, --socket=PATH         path to the unix socket\n"
-"                            (default '"SOCKET_PATH"')\n"
+"                            (default '" APPLICATION_RUNTIME_PREFIX "/tmp/nbd-%s')\n"
 "  -e, --shared=NUM          device can be shared by NUM clients (default '1')\n"
 "  -t, --persistent          don't exit on the last connection\n"
 "  -v, --verbose             display extra debugging information\n"
@@ -926,7 +925,7 @@
 
     if (device != NULL && sockpath == NULL) {
         sockpath = g_malloc(128);
-        snprintf(sockpath, 128, SOCKET_PATH, basename(device));
+        snprintf(sockpath, 128, APPLICATION_RUNTIME_PREFIX "/tmp/nbd-%s", basename(device));
     }
 
     server = qio_net_listener_new();
diff -uNr qemu-6.1.0/slirp/src/slirp.c qemu-6.1.0.mod/slirp/src/slirp.c
--- qemu-6.1.0/slirp/src/slirp.c	2021-08-24 20:36:28.000000000 +0300
+++ qemu-6.1.0.mod/slirp/src/slirp.c	2022-02-02 14:14:47.629030307 +0200
@@ -234,9 +234,12 @@
         return 0;
     }
     old_stat = *cached_stat;
-    if (stat(RESOLV_CONF_PATH, cached_stat) != 0) {
+    char *prefixed_resolvconf_path = android_dynamic_path_prefix(RESOLV_CONF_PATH);
+    if (stat(prefixed_resolvconf_path, cached_stat) != 0) {
+        free(prefixed_resolvconf_path);
         return -1;
     }
+    free(prefixed_resolvconf_path);
     if (cached_stat->st_dev == old_stat.st_dev &&
         cached_stat->st_ino == old_stat.st_ino &&
         cached_stat->st_size == old_stat.st_size &&
@@ -262,9 +265,13 @@
     unsigned if_index;
 
     assert(sizeof(tmp_addr) >= addrlen);
-    f = fopen(RESOLV_CONF_PATH, "r");
-    if (!f)
+    char *prefixed_resolvconf_path = android_dynamic_path_prefix(RESOLV_CONF_PATH);
+    f = fopen(prefixed_resolvconf_path, "r");
+    if (!f) {
+        free(prefixed_resolvconf_path);
         return -1;
+    }
+    free(prefixed_resolvconf_path);
 
     DEBUG_MISC("IP address of your DNS(s):");
     while (fgets(buff, 512, f) != NULL) {
diff -uNr qemu-6.1.0/tcg/tcg.c qemu-6.1.0.mod/tcg/tcg.c
--- qemu-6.1.0/tcg/tcg.c	2021-08-24 20:35:41.000000000 +0300
+++ qemu-6.1.0.mod/tcg/tcg.c	2022-02-02 14:14:47.629030307 +0200
@@ -4684,7 +4684,10 @@
     /* Enable this block to be able to debug the ELF image file creation.
        One can use readelf, objdump, or other inspection utilities.  */
     {
-        FILE *f = fopen("/tmp/qemu.jit", "w+b");
+        char *qemujit_tmp_file = android_dynamic_path_prefix("/tmp/qemu.jit");
+        FILE *f = fopen(qemujit_tmp_file, "w+b");
+        free(qemujit_tmp_file);
+
         if (f) {
             if (fwrite(img, img_size, 1, f) != img_size) {
                 /* Avoid stupid unused return value warning for fwrite.  */
diff -uNr qemu-6.1.0/util/module.c qemu-6.1.0.mod/util/module.c
--- qemu-6.1.0/util/module.c	2021-08-24 20:35:41.000000000 +0300
+++ qemu-6.1.0.mod/util/module.c	2022-02-02 14:14:34.180946305 +0200
@@ -280,7 +280,7 @@
     version_dir = g_strcanon(g_strdup(QEMU_PKGVERSION),
                              G_CSET_A_2_Z G_CSET_a_2_z G_CSET_DIGITS "+-.~",
                              '_');
-    dirs[n_dirs++] = g_strdup_printf("/var/run/qemu/%s", version_dir);
+    dirs[n_dirs++] = g_strdup_printf(APPLICATION_RUNTIME_PREFIX "/var/run/qemu/%s", version_dir);
 #endif
 
     assert(n_dirs <= ARRAY_SIZE(dirs));
diff -uNr qemu-6.1.0/util/qemu-sockets.c qemu-6.1.0.mod/util/qemu-sockets.c
--- qemu-6.1.0/util/qemu-sockets.c	2021-08-24 20:35:41.000000000 +0300
+++ qemu-6.1.0.mod/util/qemu-sockets.c	2022-02-02 14:14:47.629030307 +0200
@@ -923,8 +923,10 @@
         path = saddr->path;
     } else {
         const char *tmpdir = getenv("TMPDIR");
-        tmpdir = tmpdir ? tmpdir : "/tmp";
+        char *prefixed_tmpdir = android_dynamic_path_prefix("/tmp");
+        tmpdir = tmpdir ? tmpdir : prefixed_tmpdir;
         path = pathbuf = g_strdup_printf("%s/qemu-socket-XXXXXX", tmpdir);
+        free(prefixed_tmpdir);
     }
 
     pathlen = strlen(path);
