diff -uNr qemu-7.0.0/meson.build qemu-7.0.0.mod/meson.build
--- qemu-7.0.0/meson.build	2022-07-09 21:46:21.246320944 +0300
+++ qemu-7.0.0.mod/meson.build	2022-07-09 21:47:11.086746759 +0300
@@ -410,7 +410,7 @@
 add_project_arguments(config_host['GLIB_CFLAGS'].split(),
                       native: false, language: ['c', 'cpp', 'objc'])
 glib = declare_dependency(compile_args: config_host['GLIB_CFLAGS'].split(),
-                          link_args: config_host['GLIB_LIBS'].split(),
+                          link_args: config_host['GLIB_LIBS'].split() + ['-liconv'],
                           version: config_host['GLIB_VERSION'])
 # override glib dep with the configure results (for subprojects)
 meson.override_dependency('glib-2.0', glib)
