diff -uNr libslirp-v4.7.0/src/slirp.c libslirp-v4.7.0.mod/src/slirp.c
--- libslirp-v4.7.0/src/slirp.c	2022-04-26 12:50:57.000000000 +0300
+++ libslirp-v4.7.0.mod/src/slirp.c	2022-07-10 00:16:36.604109546 +0300
@@ -376,9 +376,12 @@
         return 0;
     }
     old_stat = *cached_stat;
-    if (stat(RESOLV_CONF_PATH, cached_stat) != 0) {
+    char *prefixed_resolvconf_path = android_dynamic_path_prefix(RESOLV_CONF_PATH);
+    if (stat(prefixed_resolvconf_path, cached_stat) != 0) {
+        free(prefixed_resolvconf_path);
         return -1;
     }
+    free(prefixed_resolvconf_path);
     if (cached_stat->st_dev == old_stat.st_dev &&
         cached_stat->st_ino == old_stat.st_ino &&
         cached_stat->st_size == old_stat.st_size &&
@@ -396,7 +399,6 @@
 {
     char buff[512];
     char buff2[257];
-    FILE *f;
     int found = 0;
     union {
         struct in_addr dns_addr;
@@ -405,12 +407,16 @@
     unsigned if_index;
 
     assert(sizeof(tmp_addr) >= addrlen);
-    f = fopen(RESOLV_CONF_PATH, "r");
-    if (!f)
-        return -1;
+
+    const char *env_dns4 = getenv("CONFIG_QEMU_DNS");
+    if (!env_dns4) env_dns4 = "1.1.1.1";
+    const char *env_dns6 = getenv("CONFIG_QEMU_DNS6");
+    if (!env_dns6) env_dns6 = "2606:4700:4700::1111";
+    snprintf(buff, 512, "nameserver %s\nnameserver %s\n", env_dns4, env_dns6);
 
     DEBUG_MISC("IP address of your DNS(s):");
-    while (fgets(buff, 512, f) != NULL) {
+    char *resolvconf_token = strtok(buff, "\n");
+    while (resolvconf_token) {
         if (sscanf(buff, "nameserver%*[ \t]%256s", buff2) == 1) {
             char *c = strchr(buff2, '%');
             if (c) {
@@ -421,6 +427,7 @@
             }
 
             if (!inet_pton(af, buff2, &tmp_addr)) {
+                resolvconf_token = strtok(NULL, "\n");
                 continue;
             }
             /* If it's the first one, set it to dns_addr */
@@ -448,8 +455,8 @@
                 DEBUG_MISC("  %s", res);
             }
         }
+        resolvconf_token = strtok(NULL, "\n");
     }
-    fclose(f);
     if (!found)
         return -1;
     return 0;
@@ -457,6 +464,7 @@
 
 int get_dns_addr(struct in_addr *pdns_addr)
 {
+#if 0
     static struct stat dns_addr_stat;
 
     if (dns_addr.s_addr != 0) {
@@ -467,6 +475,7 @@
             return ret;
         }
     }
+#endif
     return get_dns_addr_resolv_conf(AF_INET, pdns_addr, &dns_addr,
                                     sizeof(dns_addr),
                                     NULL, NULL, &dns_addr_time);
@@ -474,6 +483,7 @@
 
 int get_dns6_addr(struct in6_addr *pdns6_addr, uint32_t *scope_id)
 {
+#if 0
     static struct stat dns6_addr_stat;
 
     if (!in6_zero(&dns6_addr)) {
@@ -487,6 +497,7 @@
             return ret;
         }
     }
+#endif
     return get_dns_addr_resolv_conf(AF_INET6, pdns6_addr, &dns6_addr,
                                     sizeof(dns6_addr),
                                     scope_id, &dns6_scope_id, &dns6_addr_time);
diff -uNr libslirp-v4.7.0/meson.build libslirp-v4.7.0.mod/meson.build
--- libslirp-v4.7.0/meson.build	2022-04-26 12:50:57.000000000 +0300
+++ libslirp-v4.7.0.mod/meson.build	2022-07-10 01:06:43.935997445 +0300
@@ -127,23 +127,12 @@
   configuration : conf
 )
 
-lib = library('slirp', sources,
-  version : lt_version,
+lib = static_library('slirp', sources,
   c_args : cargs,
-  link_args : vflag,
-  link_depends : mapfile,
   dependencies : [glib_dep, platform_deps],
-  install : install_devel or get_option('default_library') == 'shared',
+  install : install_devel,
 )
 
-pingtest = executable('pingtest', 'test/pingtest.c',
-  link_with: [ lib ],
-  include_directories: [ 'src' ],
-  dependencies : [ platform_deps ]
-)
-
-test('ping', pingtest)
-
 if install_devel
   install_headers(['src/libslirp.h'], subdir : 'slirp')
 
